radio_beam/commonbeam.py

```diff
@@ -0,0 +1,522 @@
+        #     deconv_prob_beam = \
+        #         trans_beam.deconvolve(small_beam,
+        #                               failure_returns_pointlike=True)
+
+        #     if (not deconv_large_beam.isfinite or not deconv_prob_beam.isfinite):
+
+        #         scale_factor = 1.001
+        #         trans_beam = Beam(major=trans_major * scale_factor,
+        #                           minor=trans_minor * scale_factor,
+        #                           pa=trans_pa)
+        #     else:
+        #         can_deconv = True
+
+        #     if num == 10:
+        #         break
+
+        #     num += 1
+
+        common_beam = trans_beam
+        return common_beam
+
+
+def boundingcircle(bmaj, bmin, bpa):
+    thisone = np.argmax(bmaj)
+    return bmaj[thisone], bmaj[thisone], bpa[thisone]
+
+
+def PtoA(bmaj, bmin, bpa):
+    A = np.zeros((2, 2))
+    A[0, 0] = np.cos(bpa)**2 / bmaj**2 + np.sin(bpa)**2 / bmin**2
+    A[1, 0] = np.cos(bpa) * np.sin(bpa) * (1 / bmaj**2 - 1 / bmin**2)
+    A[0, 1] = A[1, 0]
+    A[1, 1] = np.sin(bpa)**2 / bmaj**2 + np.cos(bpa)**2 / bmin**2
+    return A
+
+
+def BinsideA(B, A):
+    try:
+        np.linalg.cholesky(B - A)
+        return True
+    except np.linalg.LinAlgError:
+        return False
+
+
+def myobjective_regularized(p, bmajvec, bminvec, bpavec):
+    # Force bmaj > bmin
+    if p[0] < p[1]:
+        return 1e30
+    # We can safely assume the common major axis is at most the
+    # largest major axis in the set
+    if (p[0] <= bmajvec).any():
+        return 1e30
+    A = PtoA(*p)
+    test = np.zeros_like(bmajvec)
+    for idx, (bmx, bmn, bp) in enumerate(zip(bmajvec, bminvec, bpavec)):
+        test[idx] = BinsideA(PtoA(bmx, bmn, bp), A)
+    obj = 1 / np.linalg.det(A)
+    if np.all(test):
+        return obj
+    else:
+        return obj * 1e30
+
+
+def common_manybeams_opt(beams, p0=None, optdict=None, verbose=False,
+                         brute=False, brute_steps=40):
+    '''
+    Optimize the common beam solution by maximizing the determinant of the
+    common beam.
+
+    ..note:: This method is experimental and requires further testing.
+
+    Parameters
+    ----------
+    beams : `~radio_beam.Beams`
+        Beams object.
+    p0 : tuple, optional
+        Initial guess parameters (`major, minor, pa`).
+    optdict : dict, optional
+        Dictionary parameters passed to `~scipy.optimize.minimize`.
+    verbose : bool, optional
+        Print the full output from `~scipy.optimize.minimize`.
+    brute : bool, optional
+        Use `~scipy.optimize.brute` to find the optimal solution.
+    brute_steps : int, optional
+        Number of positions to sample in each parameter (3).
+
+    Returns
+    -------
+    com_beam : `~radio_beam.Beam`
+        Common beam.
+    '''
+
+    raise NotImplementedError("This method is not fully tested. Remove this "
+                              "line for testing purposes.")
```

Comment:
Let me know if you want to merge with this error still present

Reply:
This can be kept. It's the best method but the minimization issues need to be solved before it can be used.